### PR TITLE
gitignore: ignore python compiled file in git status

### PR DESCRIPTION
**Pull request by aabadie**
Reviewing #7367 gave me the idea of this PR: I tried the testrunner script for the release and have the pyc file returned with git status. This PR fixes this.

### CODE REVIEW

**Comment by miri64** [comment]
Why not put into `.gitignore` in root?

**Comment by aabadie** [comment]
> Why not put into .gitignore in root?

Yes, I thought of that afterwards and also thinks it's better. Changed.

File: .gitignore [hunk 1]
```diff
@@ -48,3 +48,5 @@ Makefile.local
 .ycm_extra_conf.pyc
 __pycache__
```
Comment (miri64) [inline .gitignore]: These too are python compiled files. Can you move it below the comment (and remove the one `pyc` file above)?

**Comment by aabadie** [comment]
@miri64, better ?

**Comment by miri64** [comment]
No...

**Comment by miri64** [comment]
This was the patch I meant:

```diff
diff --git a/.gitignore b/.gitignore
index aff65bf..20f4acf 100644
--- a/.gitignore
+++ b/.gitignore
@@ -45,6 +45,8 @@ Makefile.local
 .clang_complete
 # YouCompleteMe (https://github.com/Valloric/YouCompleteMe)
 .ycm_extra_conf.py
-.ycm_extra_conf.pyc
-__pycache__
 .dlcache
+
+# Python compiled files
+*.pyc
+__pycache__
```

**Comment by aabadie** [comment]
`__pycache__` is not required in the gitignore file if you already exclude pyc files, since those folders only contains this know if files.

**Comment by miri64** [comment]
Do you have a source on that? Regardless, `.dlcache` is not a python compiled file and should stay above that comment.

**Comment by aabadie** [comment]
> Do you have a source on that?

See [PEP3147](https://www.python.org/dev/peps/pep-3147/)

**Comment by aabadie** [comment]
@miri64, if you are fine with my latest change (put dlcache in a separate block with comment), feel free to squash the commits in my branch: I'll be computerless (and git less) until mid-august ;)

**Comment by miri64** [comment]
Done via the GitHub interface (that's why there is no merge commit :/)

**Pull request merged**